jpsreport/general/ArgumentParser.cpp

```diff
@@ -418,12 +418,26 @@ bool ArgumentParser::ParseIniFile(const string & inifile)
             int num_verteces = 0;
             for(TiXmlElement * xVertex = xMeasurementArea_B->FirstChildElement("vertex"); xVertex;
                 xVertex                = xVertex->NextSiblingElement("vertex")) {
-                double box_px = xmltof(xVertex->Attribute("x")) * M2CM;
-                double box_py = xmltof(xVertex->Attribute("y")) * M2CM;
-                boost::geometry::append(poly, boost::geometry::make<point_2d>(box_px, box_py));
-                Log->Write(
-                    "\t\tMeasure area points  < %.3f, %.3f>", box_px * CMtoM, box_py * CMtoM);
-                num_verteces++;
+                // Note: Attributes are optional, their existence needs to be checked since xmltof returns 0.0 if unknown
+                if(xVertex->Attribute("x") != NULL and xVertex->Attribute("y") != NULL) {
```

Comment:
Better check against `nullptr` to explicit state what is checked here. `xVertex->Attribute("foo")` returns a `char *` but `NULL` can be either 
an integer literal with value zero or prvalue of type std::nullptr_t.

A more detailed explanation can be found here: https://cpp4arduino.com/2018/10/26/why-cpp-programmers-dont-use-null.html

Reply:
Thanks! Done.